lucene/benchmark/src/java/org/apache/lucene/benchmark/byTask/feeds/TrecContentSource.java

```diff
@@ -0,0 +1,334 @@
+ * <ul>
+ * <li><b>work.dir</b> - specifies the working directory. Required if "docs.dir"
+ * denotes a relative path (<b>default=work</b>).
+ * <li><b>docs.dir</b> - specifies the directory where the TREC files reside.
+ * Can be set to a relative path if "work.dir" is also specified
+ * (<b>default=trec</b>).
+ * <li><b>trec.doc.parser</b> - specifies the {@link TrecDocParser} class to use for
+ * parsing the TREC documents content (<b>default=TrecGov2Parser</b>).
+ * <li><b>html.parser</b> - specifies the {@link HTMLParser} class to use for
+ * parsing the HTML parts of the TREC documents content (<b>default=DemoHTMLParser</b>).
+ * <li><b>content.source.encoding</b> - if not specified, ISO-8859-1 is used.
+ * <li><b>content.source.excludeIteration</b> - if true, do not append iteration number to docname
+ * </ul>
+ */
+public class TrecContentSource extends ContentSource {
+
+  static final class DateFormatInfo {
+    DateFormat[] dfs;
+    ParsePosition pos;
+  }
+
+  public static final String DOCNO = "<DOCNO>";
+  public static final String TERMINATING_DOCNO = "</DOCNO>";
+  public static final String DOC = "<DOC>";
+  public static final String TERMINATING_DOC = "</DOC>";
+
+  /** separator between lines in the byffer */ 
+  public static final String NEW_LINE = System.getProperty("line.separator");
+
+  private static final String DATE_FORMATS [] = {
+       "EEE, dd MMM yyyy kk:mm:ss z",   // Tue, 09 Dec 2003 22:39:08 GMT
+       "EEE MMM dd kk:mm:ss yyyy z",    // Tue Dec 09 16:45:08 2003 EST
+       "EEE, dd-MMM-':'y kk:mm:ss z",   // Tue, 09 Dec 2003 22:39:08 GMT
+       "EEE, dd-MMM-yyy kk:mm:ss z",    // Tue, 09 Dec 2003 22:39:08 GMT
+       "EEE MMM dd kk:mm:ss yyyy",      // Tue Dec 09 16:45:08 2003
+       "dd MMM yyyy",                   // 1 March 1994
+       "MMM dd, yyyy",                  // February 3, 1994
+       "yyMMdd",                        // 910513
+       "hhmm z.z.z. MMM dd, yyyy",       // 0901 u.t.c. April 28, 1994
+  };
+
+  private ThreadLocal<DateFormatInfo> dateFormats = new ThreadLocal<>();
+  private ThreadLocal<StringBuilder> trecDocBuffer = new ThreadLocal<>();
+  private Path dataDir = null;
+  private ArrayList<Path> inputFiles = new ArrayList<>();
+  private int nextFile = 0;
+  // Use to synchronize threads on reading from the TREC documents.
+  private Object lock = new Object();
+
+  // Required for test
+  BufferedReader reader;
+  int iteration = 0;
+  HTMLParser htmlParser;
+  
+  private boolean excludeDocnameIteration;
+  private TrecDocParser trecDocParser = new TrecGov2Parser(); // default
+  ParsePathType currPathType; // not private for tests
+  
+  private DateFormatInfo getDateFormatInfo() {
+    DateFormatInfo dfi = dateFormats.get();
+    if (dfi == null) {
+      dfi = new DateFormatInfo();
+      dfi.dfs = new SimpleDateFormat[DATE_FORMATS.length];
+      for (int i = 0; i < dfi.dfs.length; i++) {
+        dfi.dfs[i] = new SimpleDateFormat(DATE_FORMATS[i], Locale.ENGLISH);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.